clippy_lints/src/find_is_some_on_strs.rs

```diff
@@ -0,0 +1,78 @@
+use crate::utils::{is_type_diagnostic_item, method_calls, snippet, span_lint_and_sugg};
+use if_chain::if_chain;
+use rustc_errors::Applicability;
+use rustc_hir::Expr;
+use rustc_lint::{LateContext, LateLintPass};
+use rustc_middle::ty;
+use rustc_session::{declare_lint_pass, declare_tool_lint};
+
+declare_clippy_lint! {
+    /// **What it does:** Recommends using `contains()` instead of `find().is_some()` on strings
+    ///
+    /// **Why is this bad?** Readability, this can be written more concisely as `_.contains(_)`
+    ///
+    /// **Known problems:** None.
+    ///
+    /// **Example:**
+    ///
+    /// ```rust
+    /// "hello world".find("world").is_some()
+    /// ```
+    /// Use instead:
+    /// ```rust
+    /// "hello world".contains("hello");
+    /// ```
+    pub FIND_IS_SOME_ON_STRS,
+    complexity,
+    "using `find()` followed by `is_some()` on strings, which is more
+    succinctly expressed as a call to `contains()`"
+}
+
+declare_lint_pass!(FindIsSomeOnStrs => [FIND_IS_SOME_ON_STRS]);
+
+impl LateLintPass<'_> for FindIsSomeOnStrs {
+    fn check_expr(&mut self, cx: &LateContext<'tcx>, expr: &'tcx Expr<'_>) {
+        let (method_names, arg_lists, method_spans) = method_calls(expr, 2);
+
+        if_chain! {
+            if let Some(last_method) = method_names.get(0);
+            if *last_method == sym!(is_some);
+            if let Some(first_method) = method_names.get(1);
+            if *first_method == sym!(find);
+            let find_args = arg_lists[1];
+            if let Some(find_caller) = find_args.get(0);
+            if let Some(find_arg) = find_args.get(1);
+            if is_string_or_str_slice(cx, find_caller);
+            if is_string_or_str_slice(cx, find_arg);
+            then {
+                let find_span = method_spans[1];
+                let msg = "called `is_some()` after calling `find()` \
+                    on a string. This is more succinctly expressed  by calling \
+                    `contains()`.".to_string();
+                let find_arg = snippet(cx, find_arg.span, "..");
+                span_lint_and_sugg(
+                    cx,
+                    FIND_IS_SOME_ON_STRS,
+                    find_span.with_hi(expr.span.hi()),
+                    &msg,
+                    "try this",
+                    format!("contains({})", find_arg),
+                    Applicability::MachineApplicable,
+                );
+            }
+        }
+    }
+}
+
+fn is_string_or_str_slice(cx: &LateContext<'_>, expr: &Expr<'_>) -> bool {
+    let is_string = is_type_diagnostic_item(cx, cx.typeck_results().expr_ty(expr).peel_refs(), sym!(string_type));
```

Comment:
Should we return quickly if this passes rather then checking the next one?

Reply:
Yes, that would be better.  I wasn't sure if `return` statements should be avoided or not.  But it sounds like using `return`s to exit early is good?

Reply:
I added the `return` statement, thanks for the suggestion!